src/core/os/amdgpu/amdgpuDevice.cpp

```diff
@@ -2050,11 +2090,26 @@ Result Device::GetSwapChainInfo(
         pSwapChainProperties->supportedUsageFlags.shaderRead    = 1;
         pSwapChainProperties->supportedUsageFlags.shaderWrite   = 1;
 
-        // Get format supported by swap chain.
-        pSwapChainProperties->imageFormatCount = static_cast<uint32>(ArrayLen(PresentableImageFormats));
+        // Get formats supported by swap chain. We have at least the 32 bpp formats.
+        pSwapChainProperties->imageFormatCount = baseFormatCount;
+
+        // Some gpu + amdgpu kms combinations do support fp16 scanout and display.
+        if (HasFP16DisplaySupport())
+        {
+            // fp16 is first slot in Presentable16BitSwizzledFormat[].
+            pSwapChainProperties->imageFormatCount++;
+        }
+
```

Comment:
Please add a PAL_ASSERT that pSwapChainProperties->imageFormatCount does not exceed the MaxPresentableImageFormat constant defined in palSwapChain.h.

Reply:
Done in force-push just now.